src/components/BloombergToFDC3/BloombergBridgeApp.tsx

```diff
@@ -0,0 +1,479 @@
+  };
+
+  /** Removes the FDC3 User Channel context listener. */
+  const tearDownContextListener = async () => {
+    if (contextListener){
+      contextListener.unsubscribe();
+      contextListener = null;
+    } else {
+      console.trace("tearDownContextListener: there was no context listener to remove...");
+    }
+  };
+
+  /** Retrieves teh user preference used as state for the Bloomberg terminal commands
+   *  then listens for changes from the user prefences panel or other coopies of the 
+   *  app.
+   */
+  const handleBloombergLinkPreferences = () => {
+    console.log("Retrieving Bloomberg link preferences...");
+    FSBL.Clients.ConfigClient.get(LINK_PREFERENCES_PATH, updateLinks);
+    FSBL.Clients.ConfigClient.addListener( LINK_PREFERENCES_PATH, updateLinks);
+  };
+
+  const updateLinks = (err: any, links: any) => {
+    if (!err) {
+      if (links.value) {
+        // get and the listener have different return values
+        links = links.value;
+      }
+      const filteredLinks = links.filter((link: any, index) => {
+        link.index = index;
+        return link.source.type == "fdc3.intent" && link.target.type == "BloombergCommand" && !link.bidirectional
+      })
+      console.log("Updating (FDC3 Intent related) links to: ", JSON.stringify(filteredLinks));
+      setLinks(filteredLinks);
+    }
+  };
+
+  /** Retrieves the component state from the workspace and uses it to restore those settings. */
+  const retrieveComponentState = () => {
+    console.log("Retrieving component state...");
+    FSBL.Clients.WindowClient.getComponentState(
+      {
+        fields: ["activeTab","selectedGroupIds"],
+      },
+      (err, state) => {
+        if (!err) {
+          const fetchedTab = state["activeTab"];
+          if (fetchedTab) {
+            changeTab(fetchedTab)
+          }
+          const fetchedGroupIds = state["selectedGroupIds"];
+          if (fetchedGroupIds) {
+            setSelectedGroupIds(fetchedGroupIds);
+          }
+        }
+      },
+    );
+  };
+
+  const editCommand = (link) => {
+    setEditLink(link);
+    setShowAddEditForm(true)
+  };
+  
+  useEffect(() => {
+    //Make the component aware of the status of the bloomerg connection
+    handleConnectionToBloomberg();
+
+    // Retrieve configuration for terminal commands and listen for changes
+    handleBloombergLinkPreferences();
+
+    //retrieve and restore any saved state in the workspace
+    retrieveComponentState();
+
+    // Get initial list of groups and setup event listeners
+    handleLaunchPadGroups();
+
+    // Listen for context broadcasts
+    setupContextListener();
+
+    return () => {
+      //remove listeners added above
+      FSBL.Clients.ConfigClient.removeListener(LINK_PREFERENCES_PATH, updateLinks);
+      bbgClient.removeGroupEventListener();
+      tearDownContextListener();
+    };
+
+  }, []);
+
+  return (
+    <div id="container">
+      {/* <div className="search">
```

Comment:
Is this code left her intentionally? If so, there should be a comment, explaining what it's doing.

Reply:
will remove